django/db/models/lookups.py

```diff
@@ -203,6 +203,9 @@ class In(BuiltinLookup):
     lookup_name = 'in'
 
     def process_rhs(self, compiler, connection):
+        db_rhs = getattr(self.rhs, 'db', None)
+        if db_rhs and db_rhs != connection.alias:
+            raise ValueError('Can\'t do subqueries with queries on different DBs.')
```

Comment:
What language can we use in this error to guide the user to using `list(inner)` if they're sure performance will be OK?

Something like:

```
raise ValueError('Subqueries not allowed across multiple DBs. `You can list(inner_query')` to do this in Python)'
```

@timgraham have you got some ideas here?